stackslib/src/chainstate/stacks/boot/signers-voting.clar

```diff
@@ -55,16 +55,14 @@
 (define-read-only (get-tally (reward-cycle uint) (round uint) (aggregate-public-key (buff 33)))
     (map-get? tally {reward-cycle: reward-cycle, round: round, aggregate-public-key: aggregate-public-key}))
 
-(define-read-only (get-signer-slots (signer-index uint) (reward-cycle uint))
-    (let ((height (reward-cycle-to-burn-height reward-cycle)))
-            (ok (at-block
-                (unwrap! (get-block-info? id-header-hash height) err-invalid-burn-block-height)
-                    (get-current-signer-slots signer-index)))))
-
 (define-read-only (get-current-signer-slots (signer-index uint))
-    (let ((details (unwrap! (unwrap-panic (contract-call? .signers stackerdb-get-signer-by-index signer-index)) err-invalid-signer-index)))
+    (let ((cycle (+ u1 (burn-height-to-reward-cycle burn-block-height))))
+      (get-signer-slots signer-index cycle)))
+
+(define-read-only (get-signer-slots (signer-index uint) (reward-cycle uint))
```

Comment:
should this be get-signer-weight instead so it doesn't get confused with returning the num-slots ?

Reply:
I'm just using the names that were already there.  I think the voting contract needs to be significantly refactored, but that's a separate PR I think